Sibling folder strategy

gaggle opened the pull request:
Hi,

I wanted to consider files in folders named the same as a given file, so this structure:
```
post.html
post/image.jpg
```

Can turn into:
```
post/index.html
post/image.jpg
```

I've extended the functionality of `linksets`, if set to `folder` then this folder strategy is used.

gaggle committed:
Test sibling folder functionality

gaggle committed:
Allow use of sibling-folder strategy

woodyrew commented:
Thanks for the PR.

Could you resolve the conflicts or create a new PR based off master?
Also add documentation to the readme please.

gaggle commented:
Hi @woodyrew, thanks for looking at the PR. I'll rebase my changes and force-push to this branch.

I note master is currently failing build, will this be fixed?

gaggle committed:
Add folder strategy to readme

gaggle commented:
Hi @woodyrew,

I've updated the branch, rebased off of master. With linting disabled the tests are passing. I no longer use this library so the change is a bit theoretical at this point... we can merge or close the PR as you wish. Thanks for making the library available BTW.

woodyrew commented:
Thanks for the PR. Segmentio have provided write access to the plugins to the Metalsmith community so we can move things forward now.
Fixed the build, it was due to a linting issue from a PR that was before the CI was introduced.
Fixed conflicts and merged changes into master.
